wolframalpha/__init__.py

```diff
@@ -1,84 +1,184 @@
-from xml.etree import ElementTree as etree
 from six.moves import urllib
+import xmltodict
 
 from . import compat
 
 compat.fix_HTTPMessage()
 
+
+class Client(object):
+    """
+    Wolfram|Alpha v2.0 client
+    
+    Pass an ID to the object upon instantiation, then
+    query Wolfram Alpha using the query method.
+    """
+    def __init__(self, app_id='Q59EW4-7K8AHE858R'):
+        self.app_id = app_id
+
+    def query(self, query, assumption=None):
+        """
+        Query Wolfram|Alpha using the v2.0 API
+        Allows for assumptions to be included.
+        See: http://products.wolframalpha.com/api/documentation.html#6
+        """
+        data = {
+                'input': query,
+                'appid': self.app_id
+        }
+        if assumption:
+            data.update({'assumption': assumption})
+        
+        query = urllib.parse.urlencode(data)
+        url = 'https://api.wolframalpha.com/v2/query?' + query
+        resp = urllib.request.urlopen(url)
+        assert resp.headers.get_content_type() == 'text/xml'
+        assert resp.headers.get_param('charset') == 'utf-8'
+        return Result(resp)
+
 class Result(object):
+    '''
+    Handles processing the response for the programmer.
+    '''
     def __init__(self, stream):
-        self.tree = etree.parse(stream)
+        self.tree = xmltodict.parse(stream, dict_constructor=dict)['queryresult']
         self._handle_error()
-
+        self.info = []
+        try:
+            self.pods = list(map(Pod, self.tree['pod']))
+            self.info.append(self.pods)
+        except KeyError:
+            self.pods = None
+        try:
+            self.assumptions = list(map(Assumption, self.tree['assumptions']))
+            self.info.append(self.assumptions)
+        except KeyError:
+            self.assumptions = None
+        try:
+            self.warnings = list(map(Warning, self.tree['warnings']))
+            self.info.append(self.warnings)
+        except KeyError:
+            self.warnings = None
+    
     def _handle_error(self):
-        error = self.tree.find('error')
-        if not error:
+        error_state = self.tree['@error']
```

Comment:
Is the 'error' attribute passed in every result? Why not follow the previous implementation and:

```
error = self.tree.get('error')
if not error: return

code = error['code']
...
```